sbin/build.sh

```diff
@@ -21,14 +21,49 @@ SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 # shellcheck source=sbin/common-functions.sh
 source "$SCRIPT_DIR/common-functions.sh"
 
-WORKING_DIR=$1
-TARGET_DIR=$2
-OPENJDK_REPO_NAME=$3
-JVM_VARIANT=${JVM_VARIANT:-server}
-OPENJDK_UPDATE_VERSION=$5
-OPENJDK_BUILD_NUMBER=$6
-OPENJDK_REPO_TAG=$7
-USER_SUPPLIED_CONFIGURE_ARGS=$8
+WORKING_DIR=""
+TARGET_DIR=""
+OPENJDK_REPO_NAME=""
+JVM_VARIANT="-server"
+OPENJDK_UPDATE_VERSION=""
+OPENJDK_BUILD_NUMBER=""
+OPENJDK_REPO_TAG=""
+USER_SUPPLIED_CONFIGURE_ARGS=""
+
+while [[ $# -gt 0 ]] && [[ ."$1" = .-* ]] ; do
+  opt="$1";
+  shift;
+  case "$opt" in
+    "--" ) break 2;;
+
+    "--source" | "-s" )
+    WORKING_DIR="$1"; shift;;
+
+    "--destination" | "-d" )
+    TARGET_DIR="$1"; shift;;
+
+    "--repository" | "-r" )
+    OPENJDK_REPO_NAME="$1"; shift;;
+
+    "--variant"  | "-jv" )
+    JVM_VARIANT="$1"; shift;;
```

Comment:
I think this means the -server default always gets overridden correct?

Reply:
If it gets passed in as an argument, which is the case the case when you're using makejdk.sh to call it, yes.